### PR TITLE
Calendar Error List + Focus Styling

### PR DESCRIPTION
**Pull request by nmakuch**
**This PR consists of:**

- Implementing the ErrorList component on the Calendar Page
- OnSubmit should send the page focus to this ErrorList should any information be faulty or missing
- This list will then display anchor links to the fields that are missing. In this case there is only one, the calendar dates selected.

| Before | After |
|--------|-------|
|  calerrorlist-before | calerrorlist |

- ErrorList Focus styling changes on RegistrationPage.js
- The Focus styling on the list now matches the yellow from other pages.
- A focusRing style was made in styles.js as well

| Before | After |
|--------|-------|
|      errorlist-register-before        |    errorlist-register-after |

### CODE REVIEW

File: web/src/components/Reminder.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const imBanner = css`
   color: ${theme.colour.black};
   display: inline-flex;
   align-items: center;
+  margin-top: 0rem;
```
Comment (pcraig3) [inline web/src/components/Reminder.js]: Does this actually make a difference? I can't find the page where it changes the layout.

If it's not changing anything, we could remove this rule?

**Comment by pcraig3** [comment]
Hey, I also ran an HTML validator on the page and we're getting this one warning it should be easy to fix. 

screen shot 2018-06-06 at 7 42 53 pm

[This is because the `<h2>` shows up whether or not there's a message to put into it](https://github.com/cds-snc/ircc-rescheduler/blob/master/web/src/components/ErrorMessage.js#L49). 
Can we make it conditional on whether there's a message or not? (like the `<p>` is?)

File: web/src/pages/CalendarPage.js [hunk 2]
```diff
@@ -14,13 +21,14 @@ import gql from 'graphql-tag'
...
 const DAY_LIMIT = 3
 
-const errorClass = css`
-  display: block;
-  color: red;
-  margin-bottom: ${theme.spacing.sm};
+const contentClass = css`
+  h2 {
+    margin-top: 0rem;
+  }
```
[the remark this answers is not in the document]
Comment (nmakuch) [reply web/src/pages/CalendarPage.js]: I tried the other way and a few others but my setting would only take effect after loading the page once, so every first viewing would have the wonky spacing.

contentClass on <Layout></Layout> was the only solution that seemed to work properly.

File: web/src/pages/CalendarPage.js [hunk 3]
```diff
@@ -103,7 +132,7 @@ class CalendarPage extends Component {
 
   render() {
     return (
-      <Layout>
+      <Layout contentClass={contentClass}>
```
[the remark this answers is not in the document]
Comment (nmakuch) [reply web/src/pages/CalendarPage.js]: view previous comment, we can discuss if you want

File: web/src/pages/CalendarPage.js [hunk 4]
```diff
@@ -139,15 +168,27 @@ class CalendarPage extends Component {
...
+                  }}
+                >
+                  <ErrorList message={submitError || ''}>
+                    {Object.keys(validate(values)).map((formId, i) => (
+                      <a href={`#${formId}`} key={i}>
+                        {labelNames(formId) ? labelNames(formId) : formId}
+                        <br />
+                      </a>
+                    ))}
+                  </ErrorList>
```
[the remark this answers is not in the document]
Comment (nmakuch) [reply web/src/pages/CalendarPage.js]: skipping this as per Eman's feedback during our meeting

File: web/src/components/Reminder.js (hunk 1 above)
Comment (nmakuch) [reply web/src/components/Reminder.js]: I think it's an old forgotten value during testing the component out, yep let's get rid of it

**Pull request merged**